File: library-core/src/main/java/com/mikepenz/fastadapter/IModelItem.java
package com.mikepenz.fastadapter;

import android.support.v7.widget.RecyclerView;

/**
 * Created by mikepenz on 03.02.15.
 */
public interface IModelItem<Model, Item extends IModelItem<?, ?, ?>, VH extends RecyclerView.ViewHolder> extends IItem<Item, VH> {

    /**
     * set's the model of the item
     *
     * @param model
     * @return
     */
    IModelItem<?, ?, ?> withModel(Model model);

    /**
     * get's the model of the item
     *
     * @return the model
     */
    Model getModel();
}

File: library-core/src/main/java/com/mikepenz/fastadapter/IModelItem.kt
package com.mikepenz.fastadapter

import androidx.recyclerview.widget.RecyclerView

/**
 * Created by mikepenz on 03.02.15.
 */
interface IModelItem<Model, VH : RecyclerView.ViewHolder> : IItem<VH> {

    /** The model of the item */
    var model: Model
}
